include/odp/api/spec/proto_stats.h

```diff
@@ -0,0 +1,131 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(C) 2021 Marvell.
+ */
+
+/**
+ * @file
+ *
+ * ODP Proto Stats
+ */
+
+#ifndef ODP_API_SPEC_PROTO_STATS_H_
+#define ODP_API_SPEC_PROTO_STATS_H_
+#include <odp/visibility_begin.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @addtogroup odp_proto_stats
+ *  @{
+ */
+
+/**
+ * @def ODP_PROTO_STATS_INVALID
+ * Invalid proto stats handle
+ */
+
+/** ODP proto stats params */
+typedef struct odp_proto_stats_param_t {
+} odp_proto_stats_param_t;
+
+/**
+ * Proto stats capabilities
+ */
+typedef struct odp_proto_stats_capability_t {
+} odp_proto_stats_capability_t;
+
+/**
+ * Initialize proto stats parameters
+ *
+ * Initialize an odp_proto_stats_param_t to its default values.
+ * By default all the statistics are disabled.
+ *
+ * @param param   Proto stats parameter pointer.
+ */
+void odp_proto_stats_param_init(odp_proto_stats_param_t *param);
+
+/**
+ * Get proto stats capability
+ *
+ * Get supported protocol statistics and metadata.
+ *
+ * @param capa Pointer where capabilities are updated.
+ */
+void odp_proto_stats_capability(odp_proto_stats_capability_t *capa);
```

Comment:
Is there need to link proto stats support to a pktio interface? Can all TX interfaces do the counting.

Normal pktio stats are calculated per interface, and the capa is per interface:
int odp_pktio_capability(odp_pktio_t pktio, odp_pktio_capability_t *capa);

Reply:
PKTIO already has a capabilitiy and config in odp_pktout_config_opt_t::proto_stats_ena which expresses capability of that pktio interface to support protocol stats.

Reply:
I think Petri's comment is valid since the capabilities may be different in different interfaces, not simply so that all interfaces that support some proto stats, support the exact same set of stats.

So, I think it would make sense to have the capability call take pktio as a parameter also in this case.

Then maybe we should also say that one can request any of the stats for a packet regardless of the capabilities, but the ones not supported by the particular pktio will not be updated. This way an app would not have to carefully set the statistics metadata in packets in an interface specific manner.